dist/profile/manifests/datadog_http_check.pp

```diff
@@ -0,0 +1,171 @@
+# Class: profile::datadog_http_check
+#
+# This class will install the necessary config to monitor jenkins infrastructure https endpoint
+# Parameters:
+#     contact
+#       Specify a list of users to notify regarding http_check alert
+#
+#     days_critical
+#     days_warning
+#        The check_certificate_expiration will instruct the check
+#        to create a service check that checks the expiration of the
+#        ssl certificate. Allow for a warning to occur when x days are
+#        left in the certificate, and alternatively raise a critical
+#        warning if the certificate is y days from the expiration date.
+#        The SSL certificate will always be validated for this additional
+#         service check regardless of the value of disable_ssl_validation
+#
+#     timeout
+#        The (optional) timeout in seconds.
+#
+#     threshold
+#     window
+#        The (optional) window and threshold parameters allow you to trigger
+#        alerts only if the check fails x times within the last y attempts
+#        where x is the threshold and y is the window.
+#
+
+class profile::datadog_http_check(
+  $contact = ['pagerduty'],
+  $days_warning = 30,
+  $days_critical = 10,
+  $timeout = 1,
+  $treshold = 3,
+  $window = 5
+){
+
+  include datadog_agent
+
+  class { 'datadog_agent::integrations::http_check':
```

Comment:
To clarify my point, it's possible to use exported resources if we continue to generate '/etc/dd-agent/conf.d/http_check.yaml' like we do now with something like profile::datadog_check.

But now the datadog_agent module provides an easy way to configure multiple sitename within one puppet resource, like I do in profile::datadog_http_check.
In that case, I don't see how to use exported resources even if I declare profile::datadog_http_check to be definition instead of a class.

The reason why I went with the profile::datadog_http_check was because I found it clearier even if it meant splitting the code of one 'profile' in several places but I can definitely implement the first solution.

Remark: Exported resource required puppetdb which is fine for production environment  (as it's already there) but it adds complexity to vagrant environment.